src/main/scala/org/clulab/wm/eidos/groundings/grounders/EidosOntologyGrounder.scala

```diff
@@ -101,27 +102,52 @@ abstract class EidosOntologyGrounder(val name: String, val domainOntology: Domai
     groundPatternsThenEmbeddings(splitText.mkString(" "), splitText, patterns, examples, embeddings)
   }
 
-  def exactMatchForPreds(splitText: Array[String], embeddings: Seq[ConceptEmbedding]): Option[MultipleOntologyGrounding] = {
+  // If there was an exact match, returns Some of a tuple including the SingleOntologyNodeGrounding and the
+  // Range of the match in the splitText so that we can tell how much of it was used.  No match results in None.
+  def exactMatchForPreds(splitText: Array[String], embeddings: Seq[ConceptEmbedding]): Option[(SingleOntologyNodeGrounding, Range)] = {
     // This looks for exact string overlap only!
-    val joinedText = splitText.mkString(" ")
-    val lowerText = joinedText.toLowerCase
-    // text contains node name
-    val overlapWithText = embeddings.filter(embedding => lowerText.contains(StringUtils.afterLast(embedding.namer.name.toLowerCase.replaceAll("_", " "), '/', true))).filter(emb => StringUtils.afterLast(emb.namer.name.toLowerCase.replaceAll("_", " "), '/', true).nonEmpty)
-    // node name contains text
-    val overlapWithNodeName = embeddings.filter(embedding => StringUtils.afterLast(embedding.namer.name.toLowerCase.replaceAll("_", " "), '/', true).contains(lowerText))
-    // get the maximal overlap and return it
-    val overlaps = overlapWithText++overlapWithNodeName
-    val maxOverlap = if (overlaps.nonEmpty) overlaps.maxBy(_.namer.name.length) else null
-    val returned = if (maxOverlap != null) Seq(maxOverlap).map(exactMatch => SingleOntologyNodeGrounding(exactMatch.namer, 1.0f)) else Seq.empty
-    Some(returned)
+    // This tuple is designed so that Seq.min gets the intended result, the one with the min negLength
+    // (or max length) and in case of ties, the min position in the sentence, so the leftmost match.
+    // The embedding.namer should not be required to break ties.  It goes along for the ride.
+    // For expediency, the word count is used rather than the letter count.
```

Comment:
Does this mean that the bigger match is the one that uses up more words rather than more letters? So "food aid transportation" would have more overlap to a "food_aid" node (2 words but only 7-8 chars) than to a "transportation" node (1 word but 14 characters)? If so, I think I agree with that approach!

Reply:
Yes and I will update the comment.  I'm not sure the answer will be better, but the wrong answer is calculated faster.